scaluq/gate/gate.hpp

```diff
@@ -124,11 +125,40 @@ constexpr GateType get_gate_type() {
 
 namespace internal {
 class GateBase : public std::enable_shared_from_this<GateBase> {
+protected:
+    UINT _target_mask, _control_mask;
+    void check_qubit_mask_within_bounds(StateVector& state_vector) const {
```

Comment:
```suggestion
    void check_qubit_mask_within_bounds(const StateVector& state_vector) const {
```
StateVectorを変えないのでconstにするほうが受け入れ先が広くて助かります。